### PR TITLE
Clean up TarHeader.GlobalHeadFormatPrefix

### LINKED ISSUE #77555: PAX GEA constructor setting the name value to the unformatted string
When constructing a Pax GEA entry, we internally call the `base` constructor by passing the formattable string `GlobalHeadFormatPrefix`:
https://github.com/dotnet/runtime/blob/434e7f6e613ca723247725151355b6c4843a7d8f/src/libraries/System.Formats.Tar/src/System/Formats/Tar/PaxGlobalExtendedAttributesTarEntry.cs#L28-L29

But it is not getting filled out anywhere yet. At that point, the name value would be mostly useless. If the user consults the `Name` property to see the value, it will be malformed/incomplete.

The place where we properly fill out the `name` field for this entry is when we the entry to the archive:

https://github.com/dotnet/runtime/blob/434e7f6e613ca723247725151355b6c4843a7d8f/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs#L902-L917

At that point, we should know the counter position number of the current GEA entry, as well as the tmp folder path, both which get formatted into that string.

So we should just set the name to empty in that `base` constructor name.

Thanks @stephentoub for finding this.

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/runtime/issues/77555

### CODE REVIEW

File: src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs [hunk 1]
```diff
@@ -893,32 +893,22 @@ private string GenerateExtendedAttributeName()
...
-            {
-                result = string.Format(GlobalHeadFormatPrefix, "/tmp", processId);
-            }
-            result += suffix;
+            ReadOnlySpan<char> tmp = Path.TrimEndingDirectorySeparator(Path.GetTempPath());
 
-            return result;
+            string result = $"{tmp}/GlobalHead.{Environment.ProcessId}.{globalExtendedAttributesEntryNumber}";
+            return result.Length >= FieldLengths.Name ?
+                string.Concat("/tmp", result.AsSpan(tmp.Length)) :
```
Comment (carlossanlop) [inline src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs:910]: Clever way of replacing the tmp path!

File: src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs [hunk 2]
```diff
@@ -893,32 +893,22 @@ private string GenerateExtendedAttributeName()
         }
 
         // Gets the special name for the 'name' field in a global extended attribute entry.
-        // Format: "%d/GlobalHead.%p/%n"
+        // Format: "%d/GlobalHead.%p.%n"
         // - %d: The path of the $TMPDIR variable, if found. Otherwise, the value is '/tmp'.
         // - %p: The current process ID.
-        // - %n: The sequence number of the global extended header record of the archive, starting at 1. In our case, since we only generate one, the value is always 1.
```
Comment (carlossanlop) [inline src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs:899]: Thanks for removing. Remnant from before we approved the class to describe PAX GEA entries.

File: src/libraries/System.Formats.Tar/src/System/Formats/Tar/PaxGlobalExtendedAttributesTarEntry.cs [hunk 3]
```diff
@@ -26,7 +25,7 @@ internal PaxGlobalExtendedAttributesTarEntry(TarHeader header, TarReader readerO
         /// <param name="globalExtendedAttributes">An enumeration of string key-value pairs that represents the metadata to include as Global Extended Attributes.</param>
         /// <exception cref="ArgumentNullException"><paramref name="globalExtendedAttributes"/> is <see langword="null"/>.</exception>
         public PaxGlobalExtendedAttributesTarEntry(IEnumerable<KeyValuePair<string, string>> globalExtendedAttributes)
-            : base(TarEntryType.GlobalExtendedAttributes, TarHeader.GlobalHeadFormatPrefix, TarEntryFormat.Pax, isGea: true)
+            : base(TarEntryType.GlobalExtendedAttributes, nameof(PaxGlobalExtendedAttributesTarEntry), TarEntryFormat.Pax, isGea: true) // Name == name of type for lack of a better temporary name until the entry is written
```
Comment (carlossanlop) [inline src/libraries/System.Formats.Tar/src/System/Formats/Tar/PaxGlobalExtendedAttributesTarEntry.cs:28]: Yep, good enough.

**Review by carlossanlop — APPROVED** [review]
Thanks for fixing.

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @stephentoub !

**Pull request merged**